From: Andreas Gruenbacher <agruen@suse.de>
Subject: Icecream has no way of detecting assembler-level includes

Icecream preprocesses c sources locally, and sends the result off
to a remote host for compiling. It does not recognize includes at
assembler level. The fix is to put the assemberincludes an a
separate .s file, which will always be assembled locally.

Signed-off-by: Andreas Gruenbacher <agruen@suse.de>

Index: linux-2.6.10/arch/x86_64/ia32/syscall32_syscall.s
===================================================================
--- /dev/null
+++ linux-2.6.10/arch/x86_64/ia32/syscall32_syscall.s
@@ -0,0 +1,13 @@
+/* 32bit VDSOs mapped into user space. */
+
+.section ".init.data","aw"
+
+syscall32_syscall:
+.incbin "arch/x86_64/ia32/vsyscall-syscall.so"
+syscall32_syscall_end:
+
+syscall32_sysenter:
+.incbin "arch/x86_64/ia32/vsyscall-sysenter.so"
+syscall32_sysenter_end:
+
+.previous
Index: linux-2.6.10/arch/x86_64/ia32/Makefile
===================================================================
--- linux-2.6.10.orig/arch/x86_64/ia32/Makefile
+++ linux-2.6.10/arch/x86_64/ia32/Makefile
@@ -4,7 +4,7 @@
 
 obj-$(CONFIG_IA32_EMULATION) := ia32entry.o sys_ia32.o ia32_ioctl.o \
 	ia32_signal.o tls32.o \
-	ia32_binfmt.o fpu32.o ptrace32.o syscall32.o
+	ia32_binfmt.o fpu32.o ptrace32.o syscall32.o syscall32_syscall.o
 
 sysv-$(CONFIG_SYSVIPC) := ipc32.o
 obj-$(CONFIG_IA32_EMULATION) += $(sysv-y)
Index: linux-2.6.10/arch/x86_64/ia32/syscall32.c
===================================================================
--- linux-2.6.10.orig/arch/x86_64/ia32/syscall32.c
+++ linux-2.6.10/arch/x86_64/ia32/syscall32.c
@@ -13,16 +13,6 @@
 #include <asm/tlbflush.h>
 #include <asm/ia32_unistd.h>
 
-/* 32bit VDSOs mapped into user space. */ 
-asm(".section \".init.data\",\"aw\"\n"
-    "syscall32_syscall:\n"
-    ".incbin \"arch/x86_64/ia32/vsyscall-syscall.so\"\n"
-    "syscall32_syscall_end:\n"
-    "syscall32_sysenter:\n"
-    ".incbin \"arch/x86_64/ia32/vsyscall-sysenter.so\"\n"
-    "syscall32_sysenter_end:\n"
-    ".previous");
-
 extern unsigned char syscall32_syscall[], syscall32_syscall_end[];
 extern unsigned char syscall32_sysenter[], syscall32_sysenter_end[];
 extern int sysctl_vsyscall32;
